### PR TITLE
pkgconfig post-build-check: Detect more problems, fix output and add e2e tests

### PR DESCRIPTION
**Pull request by autoantwort**
Currently not all wrong placed files are detected. Also the output was broken after the localization. Added e2e tests to verify the new check

### CODE REVIEW

File: include/vcpkg/base/messages.h [hunk 1]
```diff
@@ -2151,10 +2151,7 @@ namespace vcpkg
         "If a directory should be populated but is not, this might indicate an error in the portfile.\n"
         "If the directories are not needed and their creation cannot be disabled, use something like this in "
         "the portfile to remove them:");
-    DECLARE_MESSAGE(PortBugRemoveEmptyDirs,
-                    (),
-                    "Only the 'empty directories left by the above renames' part should be translated",
-                    "vcpkg_fixup_pkgconfig()\nfile(REMOVE_RECURSE empty directories left by the above renames)");
+    DECLARE_MESSAGE(PortBugRemoveEmptyDirs, (), "", "empty directories left by the above renames");
```
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:2154]: I don't think this is actually an improvement: it is important for translators to be able to see the context for the message, and it isn't in general safe to just combine things after the fact.

I think the correct fix is just to delete the:

```
msg::write_unlocalized_text_to_stdout(Color::warning, " vcpkg_fixup_pkgconfig()\nfile(REMOVE_RECURSE ");
```

line instead.

**Review by BillyONeal — CHANGES_REQUESTED** [review]
Looks good except for the one nitpick. Thanks!

File: include/vcpkg/base/messages.h (hunk 1 above)
Comment (autoantwort) [reply include/vcpkg/base/messages.h]: It is now `file(REMOVE_RECURSE empty directories left by the above renames)`

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the fixes!

**Pull request merged**